gunicorn/datadog_checks/gunicorn/gunicorn.py

```diff
@@ -111,7 +111,13 @@ def _count_workers(self, worker_procs):
     def _get_master_proc_by_name(self, name, tags):
         """ Return a psutil process for the master gunicorn process with the given name. """
         master_name = GUnicornCheck._get_master_proc_name(name)
-        master_procs = [p for p in psutil.process_iter() if p.cmdline() and p.cmdline()[0] == master_name]
+        master_procs = []
+        for p in psutil.process_iter():
+            try:
+                if p.cmdline() and p.cmdline()[0] == master_name:
+                    master_procs.append(p)
+            except psutil.Error:
+                continue
```

Comment:
1) Should we cover this case with a unit test?

2) `p.cmdline()` should we avoid calling that method twice?

3) Should we log something (debug/warn/info) after catching `psutil.Error`?

Feedback with only questions :)

Reply:
Just to give a better understanding, on mac some process are protected and running the `p.cmdline()` will fail. So I think we don't care about logging this, this is expected to fail with some root processes. I also think that having a unit case for this is a bit overkill, knowing we have to mock `p.cmdline()` to throw an error and just make sure the error is caught, the benefits is null I think.

I like `2)` though, I'll do something :)

Reply:
The possible issue about not logging here is that, it could be difficult to debug if `p.cmdline()` fail due to something else than a protected process.

Reply:
Right now we have no unit tests with gunicorn, a second ongoing PR is about adding tests to the gunicorn integration, this can be a good place to test more.

I better see your point, this might log a lot of thing on OSX though. A debug line is worth adding then